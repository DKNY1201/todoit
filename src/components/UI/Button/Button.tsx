import * as React from 'react';

interface IButton {
    label: string;
    id: string;
    style?: string;
    clicked?: () => void;
}

const Button = ({label, style, clicked, id}: IButton) => {
    return (
        <div id={id} className={`button-container ${style}`}>
            <button onClick={clicked}>{label}</button>
        </div>
    );
}

export default Button;