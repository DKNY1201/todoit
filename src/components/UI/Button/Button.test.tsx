import * as enzyme from 'enzyme';
import * as React from 'react';

import Button from './Button';

describe('<Button />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = enzyme.shallow(<Button label="Add Todo" style="highlight" id="add-todo-btn"/>);
    });

    it('should render a icon correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a button with provided label', () => {
        expect(wrapper.find('button').text()).toEqual('Add Todo');
    });

    it('should have corresponding class with provided style', () => {
        expect(wrapper.hasClass('highlight')).toEqual(true);
    });

    it('should have corresponding id with provided id', () => {
        expect(wrapper.hasAttribute('id')).toEqual("add-todo-btn");
    });

    it('should call provided function when click', () => {
        const clicked = jest.fn();
        wrapper.setProps({clicked});
        expect(clicked).toHaveBeenCalledTimes(0);
        wrapper.find('button').simulate('click');
        expect(clicked).toHaveBeenCalledTimes(1);
    });
});