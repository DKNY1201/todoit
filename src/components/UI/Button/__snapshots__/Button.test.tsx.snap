// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render a icon correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    id="add-todo-btn"
    label="Add Todo"
    style="highlight"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        onClick={undefined}
>
        Add Todo
</button>,
      "className": "button-container highlight",
      "id": "add-todo-btn",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Add Todo",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": "Add Todo",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          onClick={undefined}
>
          Add Todo
</button>,
        "className": "button-container highlight",
        "id": "add-todo-btn",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add Todo",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Add Todo",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
